modules/programs/pls.nix

```
@@ -0,0 +1,34 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.pls;
+
+  aliases = {
+    ls = "${pkgs.pls}/bin/pls";
+    ll =
+      "${pkgs.pls}/bin/pls -d perms -d user -d group -d size -d mtime -d git";
```

```suggestion
    ls = "${cfg.package}/bin/pls";
    ll =
      "${cfg.package}/bin/pls -d perms -d user -d group -d size -d mtime -d git";
```

Generally, my only issue with aliases and commands like these is that they might be garbage collected if the shell session (or terminal) lives long enough for the user (or a scheduled service) to run the `nix-collect-garbage` and invalidate the aliases.

I don't think HM has any good solution to this problem sadly.

---

I've used these kind of aliases (from exa) for quite a while, and they've never been garbage collected from my .bashrc. Are you sure that's a thing?